src/components/domain/CardNumberContainer/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { useRef } from 'react';
+
+import { Input, Dash } from 'components/common';
+
+import { useInputFocusing } from 'hooks';
+
+import { MAX_LENGTH } from 'constants/card';
+import type { CardNumber } from 'types/card';
+
+interface CardNumberContainerProps {
+  cardNumber: CardNumber;
+  handleChangeCardNumber: React.ChangeEventHandler<HTMLInputElement>;
+}
+
+function CardNumberContainer({ cardNumber, handleChangeCardNumber }: CardNumberContainerProps) {
+  const { num1, num2, num3, num4 } = cardNumber;
+
+  const num1Ref = useRef<HTMLInputElement>(null);
+  const num2Ref = useRef<HTMLInputElement>(null);
+  const num3Ref = useRef<HTMLInputElement>(null);
+  const num4Ref = useRef<HTMLInputElement>(null);
```

Comment:
이렇게도 될 수 있을 것 같아요! ref 객체가 많아질 때 자주 사용한답니다😊

```suggestion
const numRefs = Array.from(4).map(createRef<HTMLInputElement>);
```

Reply:
오오 좋은 방법 감사합니다!